File: app/src/main/java/com/anagha/assignment/adapter/ProductsAdapter.java
package com.anagha.assignment.adapter;

import android.content.Context;
import android.graphics.Bitmap;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.anagha.assignment.R;
import com.anagha.assignment.models.Row;
import com.nostra13.universalimageloader.core.DisplayImageOptions;
import com.nostra13.universalimageloader.core.ImageLoader;

import java.util.List;

public class ProductsAdapter extends RecyclerView.Adapter<ProductsAdapter.ProductsViewHolder> {
    List<Row> data;
    private DisplayImageOptions options;

    public ProductsAdapter(List<Row> productReviewsListForAdapter) {
        data = productReviewsListForAdapter;
        options = new DisplayImageOptions.Builder()
                .showImageOnLoading(R.drawable.ic_launcher_foreground)
                .showImageForEmptyUri(R.drawable.ic_launcher_foreground)
                .showImageOnFail(R.drawable.ic_launcher_foreground)
                .cacheInMemory(true)
                .cacheOnDisk(true)
                .considerExifParams(true)
                .bitmapConfig(Bitmap.Config.RGB_565)
                .build();
    }

    @NonNull
    @Override
    public ProductsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ProductsViewHolder myViewHolder;
        View itemLayoutView = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.list_row, parent, false);
        myViewHolder = new ProductsViewHolder(itemLayoutView);
        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ProductsViewHolder myHolder, int position) {
        ProductsViewHolder myViewHolderview = myHolder;
        if (getItem(position).getTitle() != null) {
            myViewHolderview.listRowNameTV.setText(getItem(position).getTitle());
        } else {
            myViewHolderview.listRowNameTV.setText("No Title");
        }
        if (getItem(position).getDescription() != null) {
            myViewHolderview.listRowDescTV.setText(Html.fromHtml(getItem(position).getDescription()));
        } else {
            myViewHolderview.listRowDescTV.setText("No Description");
        }
        if (getItem(position).getImageHref() != null) {
            try {
                ImageLoader.getInstance().displayImage(getItem(position).getImageHref(), myViewHolderview.listRowIM, options);
            } catch (NullPointerException npe) {
                ImageLoader.getInstance().displayImage(String.valueOf(R.drawable.ic_launcher_foreground), myViewHolderview.listRowIM, options);            }
        } else {

        }
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    private Row getItem(int position) {
        return data.get(position);
    }

    class ProductsViewHolder extends RecyclerView.ViewHolder {
        TextView listRowNameTV;
        TextView listRowDescTV;
        ImageView listRowIM;

        public ProductsViewHolder(View itemView) {
            super(itemView);
            listRowNameTV = (TextView) itemView.findViewById(R.id.list_row_nameTV);
            listRowDescTV = (TextView) itemView.findViewById(R.id.list_row_descTV);
            listRowIM = (ImageView) itemView.findViewById(R.id.list_row_IM);
        }
    }
}

File: app/src/main/java/com/anagha/assignment/adapter/ProductsAdapter.kt
package com.anagha.assignment.adapter

import android.graphics.Bitmap
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.anagha.assignment.R
import com.anagha.assignment.models.Row
import com.nostra13.universalimageloader.core.DisplayImageOptions
import com.nostra13.universalimageloader.core.ImageLoader

class ProductsAdapter(internal var data: List<Row>) : RecyclerView.Adapter<ProductsAdapter.ProductsViewHolder>() {
    private val options: DisplayImageOptions = DisplayImageOptions.Builder()
            .showImageOnLoading(R.drawable.ic_launcher_foreground)
            .showImageForEmptyUri(R.drawable.ic_launcher_foreground)
            .showImageOnFail(R.drawable.ic_launcher_foreground)
            .cacheInMemory(true)
            .cacheOnDisk(true)
            .considerExifParams(true)
            .bitmapConfig(Bitmap.Config.RGB_565)
            .build()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProductsViewHolder {
        val myViewHolder: ProductsViewHolder
        val itemLayoutView = LayoutInflater.from(parent.context).inflate(
                R.layout.list_row, parent, false)
        myViewHolder = ProductsViewHolder(itemLayoutView)
        return myViewHolder
    }

    override fun onBindViewHolder(myHolder: ProductsViewHolder, position: Int) {
        if (getItem(position).title != null) {
            myHolder.listRowNameTV.text = getItem(position).title
        } else {
            myHolder.listRowNameTV.text = "No Title"
        }
        if (getItem(position).description != null) {
            myHolder.listRowDescTV.text = Html.fromHtml(getItem(position).description)
        } else {
            myHolder.listRowDescTV.text = "No Description"
        }
        if (getItem(position).imageHref != null) {
            try {
                ImageLoader.getInstance().displayImage(getItem(position).imageHref, myHolder.listRowIM, options)
            } catch (npe: NullPointerException) {
                ImageLoader.getInstance().displayImage(R.drawable.ic_launcher_foreground.toString(), myHolder.listRowIM, options)
            }

        }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    private fun getItem(position: Int): Row {
        return data[position]
    }

    inner class ProductsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var listRowNameTV: TextView
        var listRowDescTV: TextView
        var listRowIM: ImageView

        init {
            listRowNameTV = itemView.findViewById<View>(R.id.list_row_nameTV) as TextView
            listRowDescTV = itemView.findViewById<View>(R.id.list_row_descTV) as TextView
            listRowIM = itemView.findViewById<View>(R.id.list_row_IM) as ImageView
        }
    }
}
